### PR TITLE
Add password in prompt cmd

### PR DESCRIPTION
**Pull request by vBlackOut**
Add auto password with sshpass

use user:password@host just username password
or 
user:password:port@host just username password and port
or 
username::port@host just username and port
for login in command

Compatible for IPv4/IPv6
require sshpass

### CODE REVIEW

File: sshuttle/ssh.py [hunk 1]
```diff
@@ -63,8 +62,32 @@ def empackage(z, name, data=None):
 
 def connect(ssh_cmd, rhostport, python, stderr, options):
     portl = []
+    password = False
+    ipv6 = False
+    ipv4 = False
```
Comment (brianmay) [inline sshuttle/ssh.py]: What are these for?

File: sshuttle/ssh.py [hunk 2]
```diff
@@ -63,8 +62,32 @@ def empackage(z, name, data=None):
...
+                portl = ['-p', str(int(l[0].split(":")[2]))]
+                password = l[0].split(":")[1]
+
+            if len(l[0].split(":")) == 2:
+                password = l[0].split(":")[1]
+
+            if len(l[0].split(":")) == 1:
+                portl = ['-p', str(int(l[1].split(':')[2]))]
+
+            username = l[0].split(":")[0]
```
Comment (brianmay) [inline sshuttle/ssh.py]: All of this seems to be getting very complicated. I think we need to separate parsing into another function. For example, maybe something like:

```
def parse_hostport(rhostport):
   ... deal with both ipv4 and ipv6 ...
   return username, password, host, port


username, password, host, port = parse_hostport(rhostport)

rhost = "{}@{}".format(username, host)
```

File: sshuttle/ssh.py [hunk 3]
```diff
@@ -75,12 +98,45 @@ def connect(ssh_cmd, rhostport, python, stderr, options):
...
+
+        elif len(l.split(':')) == 1:
+            l = (rhostport or '').rsplit(':', 1)
+            try:
+                rhost = l[0]
+            except KeyError:
+                rhost = l[1]
+
+        # if len(l) > 2:
+        #     portl = ['-p', str(int(l[1]))]
```
Comment (brianmay) [inline sshuttle/ssh.py]: Is this needed?

Comment (vBlackOut) [reply sshuttle/ssh.py]: try my last version please :)

File: sshuttle/ssh.py (hunk 2 above)
Comment (vBlackOut) [reply sshuttle/ssh.py]: just rerused old method

**Comment by rjharmon** [comment]
fails when run with `-v -r gateway:somePortNum ...subnets`

**Comment by vBlackOut** [comment]
exemple use: sudo sshuttle -l 0.0.0.0 --no-latency-control -vvr [**username**]:[**password**]:[**port**]@[**host**] 0/0

**Comment by rjharmon** [comment]
For the sake of backward compatibility with all existing installations, I suggest supporting `host:port` syntax.  

Also @vBlackOut by adjusting your change to use `user:pass@host:port` format, it would allow `user@host:port` to work - which would be more intuitive syntax IMO, for the no-password-specified case.

**Comment by vBlackOut** [comment]
Great suggest :) i look and repush the change thank

**Pull request merged**